audience/src/main/java/org/entcore/audience/view/dao/ViewDaoImpl.java

```diff
@@ -0,0 +1,139 @@
+package org.entcore.audience.view.dao;
+
+import fr.wseduc.webutils.Either;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.entcore.audience.view.model.ResourceViewCounter;
+import org.entcore.audience.view.model.ResourceViewDetails;
+import org.entcore.audience.view.model.ViewsCounterPerProfile;
+import org.entcore.common.sql.ISql;
+import org.entcore.common.sql.SqlResult;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class ViewDaoImpl implements ViewDao {
+
+  private final ISql sql;
+
+  public ViewDaoImpl(ISql sql) {
+    this.sql = sql;
+  }
+
+  @Override
+  public Future<Void> registerView(String module, String resourceType, String resourceId, String userId, String type) {
+    Promise<Void> promise = Promise.promise();
+
+    JsonArray params = new JsonArray();
+    params.add(module);
+    params.add(resourceType);
+    params.add(resourceId);
+    params.add(type);
+    params.add(userId);
+    params.add(LocalDateTime.now().toString());
+
+    final String query = "insert into audience.views (module, resource_type, resource_id, profile, user_id, last_view, counter) " +
+        "values (?, ?, ?, ?, ?, ?, 1) " +
+        "on conflict on constraint views_unique_constraint " +
+        "do update " +
+        "set counter = views.counter + 1, " +
+        "last_view = excluded.last_view " +
+        "where date_trunc('minute', views.last_view) <> date_trunc('minute', excluded.last_view);";
+
+    sql.prepared(query, params, results -> {
+      final Either<String, JsonArray> validatedResult = SqlResult.validResults(results);
+      if (validatedResult.isRight()) {
+        promise.complete();
+      } else {
+        promise.fail(validatedResult.left().getValue());
+      }
+    });
+    return promise.future();
+  }
+
+  @Override
+  public Future<List<ResourceViewCounter>> getCounts(String module, String resourceType, Set<String> resourceIds) {
+    final Promise<List<ResourceViewCounter>> promise = Promise.promise();
+    if(CollectionUtils.isEmpty(resourceIds)) {
+      promise.complete(Collections.emptyList());
+    } else {
+      final JsonArray params = new JsonArray();
+      params.add(module);
+      params.add(resourceType);
+
+      StringBuilder resourceIdsPlaceholder = new StringBuilder();
+      for (String resourceId : resourceIds) {
+        resourceIdsPlaceholder.append("?,");
+        params.add(resourceId);
+      }
+      resourceIdsPlaceholder.deleteCharAt(resourceIdsPlaceholder.length() - 1);
```

Comment:
ca vaut le coup de factoriser et/ou exploiter les helper dans entcore/commons/sql/SqlClient pour gérer les valeurs pour les requêtes de type IN?